kitsune/users/migrations/0035_batch_delete_inactive_users.py

```diff
@@ -0,0 +1,131 @@
+from datetime import timedelta
+import time
+import importlib
+
+from django.db import migrations
+from django.db.models import Q
+from django.utils import timezone
+from django.contrib.auth import get_user_model
+
+from kitsune.users.utils import delete_user_pipeline
+
+
+def delete_inactive_users(apps, schema_editor):
+    """
+    Delete users who haven't logged in for over three years.
+    """
+    User = get_user_model()
+    
+    has_content_criteria = (
+        Q(answer_votes__isnull=False)
+        | Q(answers__isnull=False)
+        | Q(award_creator__isnull=False)
+        | Q(badge__isnull=False)
+        | Q(created_revisions__isnull=False)
+        | Q(gallery_images__isnull=False)
+        | Q(gallery_videos__isnull=False)
+        | Q(outbox__isnull=False)
+        | Q(poll_votes__isnull=False)
+        | Q(post__isnull=False)
+        | Q(question_votes__isnull=False)
+        | Q(questions__isnull=False)
+        | Q(readied_for_l10n_revisions__isnull=False)
+        | Q(reviewed_revisions__isnull=False)
+        | Q(thread__isnull=False)
+        | Q(wiki_post_set__isnull=False)
+        | Q(wiki_thread_set__isnull=False)
+        | Q(locales_leader__isnull=False)
+        | Q(locales_reviewer__isnull=False)
+        | Q(locales_editor__isnull=False)
+        | Q(wiki_contributions__isnull=False)
+    )
+    
+    cutoff_date = timezone.now() - timedelta(days=3*365)
+    
+    query = User.objects.filter(last_login__lt=cutoff_date).annotate(has_content=has_content_criteria)
```

Comment:
Why are you using `annotate` here instead of `filter/exclude/Exists`?

Reply:
This use case absolutely works - I think it's just not specifically called out in the 4.2 docs but is in later docs.  I can run this query and see the output and it builds valid sql which executes properly. 

I'm annotating here so we can later divide our users into those with and those without content, so we can execute a quicker delete process on non-content users.